waterlog/src/Components/AlertTable.js

```diff
@@ -5,43 +5,50 @@ import { connect } from 'react-redux';
 import { fetchSegmentsLeaks } from '../actions/SegmentLeaksActions';
 
 class AlertTableComponent extends Component {
-	componentWillMount() {
+	componentDidMount() {
 		this.props.fetchSegmentsLeaks();
 	}
+
 	alertInfo = [ 'DATE', 'DESCRIPTION', 'COST', 'STATUS' ];
 
+	getStatusIcon = function(severity) {
+		switch (severity) {
```

Comment:
If you going to compare to strings ensure that the receiving string is in lower Case/Upper Case and is trimmed. The same apply for the string it being compared to.

Reply:
Alternatively, make use of ENUMS but @RetroGuillaume solution does suffice

Reply:
Disclaimer, I believe if you are using js es6 then you can use enums otherwise do not 
https://stackoverflow.com/questions/44447847/enums-in-javascript-with-es6

Reply:
converted severity to lower case